examples/src/main/java/com/mapbox/navigation/examples/core/SilentWaypointsRerouteActivity.kt

```diff
@@ -0,0 +1,377 @@
+    private fun initListeners() {
+        setupReplayControls()
+        btnStartNavigation.setOnClickListener {
+            updateCameraOnNavigationStateChange(true)
+            navigationMapboxMap?.addProgressChangeListener(mapboxNavigation!!)
+            if (mapboxNavigation?.getRoutes()?.isNotEmpty() == true) {
+                navigationMapboxMap?.startCamera(mapboxNavigation?.getRoutes()!![0])
+            }
+            mapboxNavigation?.registerRouteProgressObserver(ReplayProgressObserver(mapboxReplayer))
+            mapboxNavigation?.startTripSession()
+            btnReroute.visibility = VISIBLE
+            btnStartNavigation.visibility = GONE
+            mapboxReplayer.play()
+        }
+        btnReroute.setOnClickListener {
+            mapboxNavigation?.getRerouteController()?.reroute(
+                object : RerouteController.RoutesCallback {
+                    override fun onNewRoutes(routes: List<DirectionsRoute>) {
+                    }
+                }
+            )
+        }
+    }
+
+    private fun setupReplayControls() {
+        seekBar.max = 4
+        seekBar.progress = 1
+        seekBarText.text = getString(R.string.replay_playback_speed_seekbar, seekBar.progress)
+        seekBar.setOnSeekBarChangeListener(
+            object : SeekBar.OnSeekBarChangeListener {
+                override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+                    mapboxReplayer.playbackSpeed(progress.toDouble())
+                    seekBarText.text = getString(R.string.replay_playback_speed_seekbar, progress)
+                }
+
+                override fun onStartTrackingTouch(seekBar: SeekBar) {}
+                override fun onStopTrackingTouch(seekBar: SeekBar) {}
+            }
+        )
+    }
+
+    private fun updateCameraOnNavigationStateChange(navigationStarted: Boolean) {
+        navigationMapboxMap?.apply {
+            if (navigationStarted) {
+                updateCameraTrackingMode(NavigationCamera.NAVIGATION_TRACKING_MODE_GPS)
+                updateLocationLayerRenderMode(RenderMode.GPS)
+            } else {
+                updateCameraTrackingMode(NavigationCamera.NAVIGATION_TRACKING_MODE_NONE)
+                updateLocationLayerRenderMode(RenderMode.COMPASS)
+            }
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        // This is not the most efficient way to preserve the route on a device rotation.
+        // This is here to demonstrate that this event needs to be handled in order to
+        // redraw the route line after a rotation.
+        directionRoute?.let {
+            outState.putString(Utils.PRIMARY_ROUTE_BUNDLE_KEY, it.toJson())
+        }
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
+        super.onRestoreInstanceState(savedInstanceState)
+        directionRoute = Utils.getRouteFromBundle(savedInstanceState)
```

Comment:
We also need to set this route to `MapboxNavigation#setRoutes`, otherwise, it won't be recognized anyway.

Reply:
And `NavigationMapboxMap#drawRoutes`.

Reply:
👍